proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java

```diff
@@ -494,7 +494,11 @@ public void run()
                             }
                             if ( server == null )
                             {
-                                server = bungee.getServerInfo( listener.getDefaultServer() );
```

Comment:
1. You call the event in line ```486``` always without the server the user is going to be if the ```setServer()``` method is called - plugins can't easily determine the target server.
2. The server selection logic is from line ```488``` to ```498``` (before your change). You should call the event after this is done and you should set the server in the event before you call the event.
3. Put your added lines ```497``` to ```503``` after the additons mentioned in 2.